pkgs/ffigen/lib/src/code_generator/func.dart

```diff
@@ -145,13 +145,15 @@ class Func extends LookUpBinding {
     }
 
     if (ffiNativeConfig.enabled) {
-      final assetString = ffiNativeConfig.assetId != null
-          ? ", assetId: '${ffiNativeConfig.assetId}'"
-          : '';
-      final isLeafString = isLeaf ? ', isLeaf:true' : '';
       final nativeFuncName = needsWrapper ? funcVarName : enclosingFuncName;
       s.write('''
-@${w.ffiLibraryPrefix}.Native<$cType>(symbol: '$originalName'$assetString$isLeafString)
+${makeNativeAnnotation(
+        w,
+        nativeType: cType,
+        differentName: originalName != nativeFuncName ? originalName : null,
```

Comment:
Move this expression into the function and just take `originalName` and `nativeFuncName` as arguments. (The expression is duplicated among call sites.)